Windows Accessibility: Report checked checked state of items in wxCheckListBox

glenrgordon opened the pull request:
The standard Windows ListBox class has no concept of checked/unchecked
state of items. WXWidgets obviously overcomes this with owner drawn
items so that the checked boxes can be drawn. But this info gets lost
when using the standard Windows IAccessible ListBox implementation
since said implementation has no concept of checked state for
individual items.

Fixes https://github.com/wxwidgets/wxwidgets/issues/25948

glenrgordon committed:
Windows Accessibility: Report checked checked state of items in wxCheckListBox

The standard Windows ListBox class has no concept of checked/unchecked
state of items. WXWidgets obviously overcomes this with owner drawn
items so that the checked boxes can be drawn. But this info gets lost
when using the standard Windows IAccessible ListBox implementation
since said implementation has no concept of checked state for
individual items.

Fixes https://github.com/wxwidgets/wxwidgets/issues/25948

vadz commented on src/common/wincmn.cpp:
```diff
@@ -3955,7 +3955,7 @@ wxAccStatus wxWindowAccessible::GetParent(wxAccessible** parent)
             return wxACC_OK;
         else
-            return wxACC_FAIL;
+            return wxACC_NOT_IMPLEMENTED;
```
Is this related to the other changes? If not, it should be ideally extracted into a separate commit, with the explanation of why is this necessary in its commit message.

vadz reviewed (commented):
Great, thanks for implementing this!

I didn't test it, but the code looks correct to me, it would just be nice to address the minor comments/questions below before merging it.

Thanks again!

glenrgordon committed:
Address PR comments.

glenrgordon commented:
Thanks for the reivew!
I think I've addressed all of the issues. Let me know if I've missed
anything.
Will add a new PR for the portion that got removed because it's  only
indirectly related to this change.
